File: typescript/numbers/test/numbers_iterative.test.ts
import { fibonacci, factorial, sum_numbers } from '../src/numbers_iterative';

describe('Iterative Number Functions', () => {
  test('fibonacci(5) should be 5', () => {
    expect(fibonacci(5)).toBe(5);
  });
  test('factorial(5) should be 120', () => {
    expect(factorial(5)).toBe(120);
  });
  test('sum_numbers(5) should be 15', () => {
    expect(sum_numbers(5)).toBe(15);
  });
});

File: _experimental/_zig/numbers/test/numbers_iterative_test.zig
const std = @import("std");
const numbers_iterative = @import("numbers_iterative");

test "iterative Fibonacci" {
    try std.testing.expect(numbers_iterative.fibonacci(5) == 5);
}

test "iterative Factorial" {
    try std.testing.expect(numbers_iterative.factorial(5) == 120);
}

test "iterative Sum of N" {
    try std.testing.expect(numbers_iterative.sum_numbers(5) == 15);
}
